### PR TITLE
Update readme with new terminology.

### PR DESCRIPTION
**Pull request by mike-kaufman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,21 +14,35 @@ and Javascript developers.
...
 
 ## Terminology
-During program execution, there are four different types of events that let us track async context:
-1.  `executeBegin` - indicates the begining of execution of an asynchronous function.
-2.  `link` - indicates a callback was queued for later asynchronous execution. 
-3.  `cause` - indicates a previously linked function was resolved. 
-4.  `executeEnd` - indicates the end of execution of an asynchronous function
+One of the key challenges with "asynchronous context" is the lack of agreed upon terminology and semantics.  Let's define some:
+
+1.  `Execution Frame` - An `Execution Frame` is a period of program execution, defined precisely as the period of time that a special function, called a `Continuation`, is executing. At a lower level of abstraction, you can think of an `Execution Frame` as the period of time from when specific call frame is pushed on the stack, until that call frame is unwound off of the stack. Not all functions are `Continuations` (more on that below).
```
Comment (mike-kaufman) [inline README.md:29]: alternative here is to call this `Continuation Frame`

**Comment by mrkmarron** [comment]
I like it -- LGTM

**Comment by kjin** [comment]
Sorry, missed this notification until now! I took a look and generally the definitions are understandable to me --  here are some thoughts/questions:

- Something I'm confused by is that it seems like the word `Point` is overloaded here -- though it seems clear to me that `Continuation Point` refers to a function, `Link Point` and `Ready Point` refer to points in program execution, so not necessarily a function? Do they refer to specific statements, or call sites, instead? (I think the concept of `continuation points` could use a word other than "point")
- Is my understanding correct that with the exception of the main program body, execution frames should be identifiable by a corresponding function?
- IIUC, "ready" is the new word for "causal context". If so this is definitely an improvement because that makes it easier to prevent misstatements :)

**Comment by mike-kaufman** [comment]
> Something I'm confused by is that it seems like the word Point is overloaded here

Yes, you're absolutely right here.  I wanted to update to a different name, but couldn't come up w/ anything that resonated.  I'm open to suggestions here.

>  `Continuation Point` refers to a function, `Link Point` and `Ready Point` refer to points in program execution, so not necessarily a function? 

Yes, exactly.  `Contnuation Point` is a function that takes a `Continuation` as a parameter, and `Link Point`/`Ready Point` are points in program execution. 

> Is my understanding correct that with the exception of the main program body, execution frames should be identifiable by a corresponding function?

An `Execution Frame` is a specific invocation of a `Continuation`.  It is possible for an `Continuation` to be invoked multiple times, resulting in multiple `Execution Frames`.

> IIUC, "ready" is the new word for "causal context".

Yup. :)

**Comment by kjin** [comment]
> An Execution Frame is a specific invocation of a Continuation. It is possible for an Continuation to be invoked multiple times, resulting in multiple Execution Frames.

That makes sense -- this would be helpful as part of the definition!

**Pull request merged**